### ISSUE TITLE
Translation system

### ISSUE THREAD

**Issue by hosios** [tbar0970/jethro-pmm#193]
This is such a great web application. But is it for no use in our church, because very few people know English. I searched for it in the repository but apparently there is nobody talking about it. 
Isn't it possible to incorporate some sort of translation for this? I guess there would be a lot of help from the community. I tried to do it locally on my computer, but I can't find all the necessary strings to translate it. I can't do this alone. There is a lot of work into it. Sometimes I don't know if it is the term from the database, or it is the html visible from the web browser. 
It would be nice if the user itself could choose the language on the login page.

**Comment by tbar0970** [comment]
It would be really good to support other languages, but a fair bit of work.

Most of the text is in the PHP or JS code, but there are a few places where we'd need to adjust the architecture a little to allow translation.

I've just sent an email to someone I know was working on translation, and invited him to comment here.

**Comment by hosios** [comment]
Thank you

**Comment by hosios** [comment]
I was looking for examples, and I really liked the system [Syncthing](https://github.com/syncthing) uses. Their pages translate **instantly.** Each language has a file, and the GUI just loads the one chosen in a drop-down menu by the user. **Brilliant.** Looking further into their github [page](https://github.com/syncthing/syncthing/tree/master/gui/default/assets/lang) The readme first.txt shows they are using Transifex. But, unfortunately it is not open source any more. So I guess that is a no go. Is there something similar we could use? 
I guess though it would mean a lot of structure changes, right? On the other hand it means a very good system that can be used by the world. And I know many churches would profit from this effort.

_Thank you anyway._

**Comment by hosios** [comment]
I found one that looks promising - [Weblate](https://github.com/nijel/weblate)

**Comment by AndresSereno** [comment]
Hi, I've received the email :)
I've applied internationalization to almost the whole solution, in order to use Jethro in our church. (Spanish speaker) 
Below a summary of what is already done, and pending points.

Done:
-Change configuration in order to set the language.  
-Update all pages adding gettext() everywhere a message or label is shown. (except admin pages) 
-Update functions used to determinate labels and menu items automatically.
-Update some js scripts in order to get Jethro working fine. Here we need to think a better solution, I just changed from "hardcoded in english" to "hardcoded in spanish" :(
-Consequently a message.mo for spanish have been generated. 

Pending:

- Apply internationalization to some pages, the ones which are under Admin menu option. (I can finish it, I need just a couple of days.)
- Solve the problem for default person status. (The "Adult" person status is required and fixed in some code conditions) PHP and JS. I guess these are the architecture changes which Tom is talking about. 
- Solve the english text messages shown from JS. Maybe adding jed, or jquery i18n pluggin.

So, I would like to share the features, we need just to manage the merge, because at the moment, we can work as a branch.

**Comment by hosios** [comment]
That sounds great.
Awesome. 
And for the time being, Spanish is exactly the language I need too. So it couldn't get any better. I'm willing to help with the translation from the admin pages, if needed.
And I'm willing to start the process with Brazilian Portuguese and German as well. 

I changed the person status from Adult to Adulto, and didn't notice any problems in the system, but I'm just starting to use it and to put in the data, and I didn't use all the features yet.

Any ways, thank you.

Looking forward to work together. 

(I'm new to github, but I think I got it, so if the branch for internationalization is started I'll be hoping in.)
